src/jit/lsrabuild.cpp

```diff
@@ -2501,7 +2506,11 @@ RefPosition* LinearScan::BuildUse(GenTree* operand, regMaskTP candidates, int mu
     Interval* interval;
     bool      regOptional = operand->IsRegOptional();
 
-    if (operand->gtRegNum != REG_NA)
+    if (candidates != RBM_NONE)
+    {
+        assert((operand->gtRegNum == REG_NA) || (candidates == genRegMask(operand->GetRegByIndex(multiRegIdx))));
```

Comment:
This assert is too strong, because there is no guarantee that the register required for a target is the same as the register required by the consumer. That said, this assert fired because `fgMorphTailCall()` puts `REG_R11` on the node that is created by `Compiler::fgGetStubAddrArg()`, but in `Lowering` the `PUTARG_REG` for this node puts the value in `REG_R8`. I'm not sure if this is what is supposed to be happening. @jashook - can you comment on this? This results in the following in codegen:
```
Generating: N071 (  3, 10) [000028] ------------        t28 =    CNS_INT(h) long   0x7ffa8aa20020 ftn REG r11 $c2
IN0010:        mov      r11, 0x7FFA8AA20020
                                                             /--*  t28    long   
Generating: N073 (???,???) [000063] ------------        t63 = *  PUTARG_REG long   REG r8
IN0011:        mov      r8, r11
```